language-plutus-core/src/Language/PlutusCore/Quote.hs

```diff
@@ -88,23 +89,28 @@ markNonFreshType
     :: (HasUnique (tyname a) TypeUnique, MonadQuote m)
     => Type tyname a
     -> m ()
-markNonFreshType = markNonFresh . maximum . collectTypeUniques
+markNonFreshType = markNonFreshMax . collectTypeUniques
 
 -- | Marks all the 'Unique's in a term as used, so they will not be generated in future. Useful if you
 -- have a term which was not generated in 'Quote'.
 markNonFreshTerm
     :: (HasUnique (tyname a) TypeUnique, HasUnique (name a) TermUnique, MonadQuote m)
     => Term tyname name a
     -> m ()
-markNonFreshTerm = markNonFresh . maximum . collectTermUniques
+markNonFreshTerm = markNonFreshMax . collectTermUniques
```

Comment:
`maximum` throws when there are no uniques, hence I added a function that defaults to `Unique 0` when there are no uniques.